### DISCUSSION TITLE
gitlab+vela 验证测试 CICD环境部署问题求教。https://kubevela.io/zh/docs/next/case-studies/gitops

### DISCUSSION THREAD

**Discussion by SUMMERLm** [kubevela/kubevela#3391] [Q&A]
_**### _gitlab环境_**_
<img width="1386" alt="image" src="https://user-images.githubusercontent.com/92348982/156998137-f13d47bc-f81b-42d8-a1d2-60b4a509bb8e.png">

**_git使用ops分支进行测试_**
[root@vela-master for-SREs]# git branch
  master
* ops

**_sample文件下载及按需调整_**
[root@vela-master for-SREs]# ls
apps  clusters  infrastructure

[root@vela-master for-SREs]# cat clusters/infra.yaml
apiVersion: core.oam.dev/v1beta1
kind: Application
metadata:
  name: infra
spec:
  components:
  - name: database-config
    type: kustomize
    properties:
      repoType: git
      **url: http://10.11.9.110:8888/lumingming/gitops.git**
      pullInterval: 10m
      git:
        branch: ops
      path: ./infrastructure
[root@vela-master for-SREs]# cat clusters/apps.yaml
apiVersion: core.oam.dev/v1beta1
kind: Application
metadata:
  name: apps
spec:
  components:
  - name: apps
    type: kustomize
    properties:
      repoType: git
      **url: http://10.11.9.110:8888/lumingming/gitops.git**
      pullInterval: 1m
      git:
        branch: ops
      path: ./apps
[root@vela-master for-SREs]#  vela list
APP               	COMPONENT         	TYPE      	TRAITS 	PHASE  	HEALTHY	STATUS   	CREATED-TIME
apps              	apps              	kustomize 	       	running	healthy	         	2022-03-07 15:30:18 +0800 CST
example-app       	hello-world-server	webservice	scaler 	running	healthy	Ready:2/2	2022-02-15 09:33:31 +0800 CST
infra             	database-config   	kustomize 	       	running	healthy	         	2022-03-07 15:30:08 +0800 CST
multi-env         	multi-env         	webservice	scaler 	running	healthy	Ready:5/5	2022-02-15 14:10:40 +0800 CST
rollout-trait-test	express-server    	webservice	rollout	running	healthy	         	2022-02-18 09:56:18 +0800 CST
website           	frontend          	webservice	       	running	healthy	Ready:1/1	2022-02-18 10:15:43 +0800 CST

[root@vela-master for-SREs]# kubectl get kustomizations.kustomize.toolkit.fluxcd.io
NAME              READY   STATUS                                    AGE
apps              False   **Source is not ready, artifact not found**   70m
database-config   False   **Source is not ready, artifact not found**   70m
[root@vela-master for-SREs]#
[root@vela-master for-SREs]# kubectl describe kustomizations.kustomize.toolkit.fluxcd.io database-config
Name:         database-config
Namespace:    default
Labels:       app.oam.dev/app-revision-hash=420459508b222579
              app.oam.dev/appRevision=infra-v1
              app.oam.dev/cluster=
              app.oam.dev/component=database-config
              app.oam.dev/name=infra
              app.oam.dev/namespace=default
              app.oam.dev/resourceType=WORKLOAD
              app.oam.dev/revision=database-config-v1
              oam.dev/render-hash=66a24b66aac8d4e7
              workload.oam.dev/type=kustomize
Annotations:  app.oam.dev/last-applied-configuration:
                {"apiVersion":"kustomize.toolkit.fluxcd.io/v1beta1","kind":"Kustomization","metadata":{"annotations":{"oam.dev/kubevela-version":"v1.2.3"}...
              oam.dev/kubevela-version: v1.2.3
API Version:  kustomize.toolkit.fluxcd.io/v1beta1
Kind:         Kustomization
Metadata:
  Creation Timestamp:  2022-03-07T07:30:08Z
  Finalizers:
    finalizers.fluxcd.io
  Generation:  3
  Managed Fields:
    API Version:  kustomize.toolkit.fluxcd.io/v1beta1
    Fields Type:  FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          .:
          f:app.oam.dev/last-applied-configuration:
          f:oam.dev/kubevela-version:
        f:labels:
          .:
          f:app.oam.dev/app-revision-hash:
          f:app.oam.dev/appRevision:
          f:app.oam.dev/cluster:
          f:app.oam.dev/component:
          f:app.oam.dev/name:
          f:app.oam.dev/namespace:
          f:app.oam.dev/resourceType:
          f:app.oam.dev/revision:
          f:oam.dev/render-hash:
          f:workload.oam.dev/type:
      f:spec:
        .:
        f:force:
        f:interval:
        f:path:
        f:prune:
        f:sourceRef:
          .:
          f:kind:
          f:name:
          f:namespace:
        f:validation:
    Manager:      kubevela
    Operation:    Update
    Time:         2022-03-07T07:30:08Z
    API Version:  kustomize.toolkit.fluxcd.io/v1beta1
    Fields Type:  FieldsV1
    fieldsV1:
      f:metadata:
        f:finalizers:
          .:
          v:"finalizers.fluxcd.io":
    Manager:      kustomize-controller
    Operation:    Update
    Time:         2022-03-07T07:30:08Z
    API Version:  kustomize.toolkit.fluxcd.io/v1beta1
    Fields Type:  FieldsV1
    fieldsV1:
      f:status:
        .:
        f:conditions:
        f:observedGeneration:
    Manager:         kustomize-controller
    Operation:       Update
    Subresource:     status
    Time:            2022-03-07T07:30:08Z
  Resource Version:  19292301
  UID:               0939d523-b35b-41f5-960c-c449acd36dc8
Spec:
  Force:     false
  Interval:  10m
  Path:      ./infrastructure
  Prune:     true
  Source Ref:
    Kind:       GitRepository
    Name:       database-config
    Namespace:  default
  Validation:   client
Status:
  Conditions:
    Last Transition Time:  2022-03-07T07:30:08Z
    **_Message:               Source is not ready, artifact not found_**
    Reason:                ArtifactFailed
    Status:                False
    Type:                  Ready
  Observed Generation:     3
Events:                    <none>

**Comment by FogDong** [comment]
如果是私有仓库的话，需要配置秘钥用于访问，参考：https://github.com/oam-dev/samples/blob/master/9.GitOps_Demo/for-developers/kubevela-config/clusters/apps.yaml#L12

以及：https://kubevela.io/zh/docs/case-studies/gitops#%E9%85%8D%E7%BD%AE%E7%A7%98%E9%92%A5%E4%BF%A1%E6%81%AF
